server/routes.py

```diff
@@ -0,0 +1,179 @@
+
+    return jsonify({"users": output})
+
+@app.route('/user/<public_id>', methods=['GET'])
+@token_required
+def get_one_user(current_user, public_id):
+    if not current_user.admin:
+        return jsonify({"message": "No permisson for the operation"})
+    user = User.query.filter_by(public_id=public_id).first()
+    if not user:
+        return jsonify({"message": "No user found!"})
+    user_data = {}
+    user_data["public_id"] = user.public_id
+    user_data["username"] = user.username
+    user_data["password"] = user.password
+    user_data["admin"] = user.admin
+
+    return jsonify({"user": user_data})
+
+@app.route('/user', methods=['POST'])
+@token_required
+def create_user(current_user):
+    if not current_user.admin:
+        return jsonify({"message": "No permisson for the operation"})
+    data = request.get_json()
+    users = User.query.all()
+    for user in users:
+        if data["username"] == user.username:
+            return jsonify({"message": "Username is aready in use"})
+    hashed_password = generate_password_hash(data["password"], method="sha256")
+    new_user = User(
+        public_id=str(uuid.uuid4()),
+        username=data["username"],
+        password=hashed_password,
+        admin=False)
+    db.session.add(new_user)
+    db.session.commit()
+    return jsonify({"message": "New user created!"})
+
+@app.route('/user/<public_id>', methods=['PATCH'])# zasto moze i PUT?
+@token_required
+def promote_user(current_user, public_id):
+    if not current_user.admin:
+        return jsonify({"message": "No permisson for the operation"})
+    user = User.query.filter_by(public_id=public_id).first()
+    if not user:
+        return jsonify({"message": "No user found!"})
+    user.admin = True
+    db.session.commit()
+    return jsonify({"message": "The user has been prometed to admin"})
+
+@app.route('/user/<public_id>', methods=['DELETE'])
+@token_required
+def delete_user(current_user, public_id):
+    if not current_user.admin:
+        return jsonify({"message": "No permisson for the operation"})
+    user = User.query.filter_by(public_id=public_id).first()
+    if not user:
+        return jsonify({"message": "No user found!"})
+    db.session.delete(user)
+    db.session.commit()
+    return jsonify({"message": "The user has been deleted!"})
+
+@app.route('/login')# ako ne navedem metod, onda je GET?
+def login():
+    auth = request.authorization
+    
+    if not auth or not auth.username or not auth.password:
+        return make_response('Could not verify', 401,
+            {'WWW-Authenticate': 'Basic realm="Login required!"'})# zasto WWW-Authenticate - neka standardna forma?
+
+    user = User.query.filter_by(username=auth.username).first()
+
+    if not user:
+        return make_response('Could not verify', 401,
+            {'WWW-Authenticate': 'Basic realm="Login required!"'})
+
+    if check_password_hash(user.password, auth.password):
+        token = jwt.encode({
```

Comment:
Use `flask_jwt_extended` library instead of PyJWT, it's far more powerful library for handling JWT stuff.